### ISSUE TITLE
Improve items list component

### ISSUE THREAD

**Issue by gtsiolis** [gitpod-io/gitpod#4635]
### Problem to solve

In https://github.com/gitpod-io/gitpod/pull/4454, we've added a reusable items list component used in _Workspaces_, _Environement Variables_, etc. In parallel, we've also fixed the issue with multiple dropdowns, see https://github.com/gitpod-io/gitpod/pull/4502. Cc @corneliusludmann 

However, there's still room for improving the UX and interaction of the items list component and the more actions dropdown.

### Proposal

The items list component could be improved by doing the following:
1. Do not remove the hover state on the more actions button when triggered and moving the cursor away. (_Dropdown B_) 
2. Do not remove the hover state of the list item when triggering the more actions and moving away the cursor. (_Dropdown C_) 

| Dropdown A | Dropdown B | Dropdown C |
|-|-|-|
| <img width="280" alt="list-item-1" src="https://user-images.githubusercontent.com/120486/123636874-08a01d80-d826-11eb-9e76-806570e2fa2e.png"> | <img width="280" alt="list-item-2" src="https://user-images.githubusercontent.com/120486/123636884-09d14a80-d826-11eb-9200-58da2d4226d2.png"> | <img width="280" alt="list-item-3" src="https://user-images.githubusercontent.com/120486/123636893-0c33a480-d826-11eb-8109-7aeb747aa554.png"> |

**Comment by gtsiolis** [comment]
This is still relevant. Adding **meta: never-stale** label.

**Comment by gtsiolis** [comment]
Cc @jldec in case we could prioritize this small fix.

Also adding this as a good-first-issue in case someone from the community would be interested in picking this up.

**Comment by jldec** [comment]
thanks @gtsiolis

**Comment by aboqasem** [comment]
Hi!

I was looking at the issue and noticed that the list is not consistent with tabs, it uses divs instead of buttons which does not respect semantic HTML.

I thought of using a [`group-focus`](https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-parent-state), but the actions button is a div as well 😅.

**Comment by gtsiolis** [comment]
Hi @aboqasem! Your [comment above](https://github.com/gitpod-io/gitpod/issues/4635#issuecomment-1152264649) comes with valid points and concerns regarding accessibility. ✔️ 

However, this is indeed a different issue that needs to be resolved regardless what HTML elements are used under the hood and is not related with the interaction issue described in this issue.

I've opened a separate issue for improving the more actions button accessibility in the context menu component, as we generally tend to work with smaller scope issues containing MVC (minimum viable changes) so that we can ship faster and plan better. Tackling these two issues in one step or separately is completely fine. 💯 
- https://github.com/gitpod-io/gitpod/issues/10586

Re-posting from https://github.com/gitpod-io/gitpod/issues/10586 to also help with answering the [question above](https://github.com/gitpod-io/gitpod/issues/4635#issuecomment-1152264649).

> Generally, a div-based button can become quite accessible, however, you'd still need to make the element focusable and define event handlers when a button-based button automatically takes care of this. See relevant documentation[[1](https://www.w3.org/WAI/ARIA/apg/patterns/button/)][[2](https://www.w3.org/WAI/ARIA/apg/example-index/button/button.html)][[3](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role)][...].

> For example, in this case where we're using a div-based button, we're missing the basic role="button" attribute, which could improve the existing situation. ➿

**Comment by henit-chobisa** [comment]
Hey @gtsiolis, I would be keen to work on this issues. 🍊

**Comment by gtsiolis** [comment]
Hey @henit-chobisa! Feel free to open a pull request and link back this issue. The proposal in the issue description should clearly define what we need to change here. Let me know if this is not clear enough. 🏓

**Comment by henit-chobisa** [comment]
Thanks for getting back to me @gtsiolis, I am clear with the issue, to confirm we need to maintain the selection of the workspace item and the button corresponding to context menu till the time the drop down is active.
I have already started working on it, will soon make a PR.

**Comment by gtsiolis** [comment]
Correct, @henit-chobisa!

**Comment by henit-chobisa** [comment]
🤩, would be awesome if you could assign me, will make a PR soon.

**Comment by gtsiolis** [comment]
Sure, @henit-chobisa! I generally avoid assigning anyone outside the team to avoid blocking community contributions to stale issues that never get a PR which is very common across open source projects. However, I like challenging this approach—assigned!

**Comment by henit-chobisa** [comment]
Thank you so much for assigning me @gtsiolis, grateful

**Comment by henit-chobisa** [comment]
@gtsiolis 
I have almost solved the issue, currently it looks like this, but for this I had to pass a function all the way from `WorkspaceEntry` component to `ContextMenu` component, because the expansion state is of the `ContextMenu` component itself, can I improve it in some way? Like setting up context for this particular element ?

https://user-images.githubusercontent.com/72302948/215723816-e1d7fcd0-9f45-4639-9a57-60d0ba981eae.mov

**Comment by gtsiolis** [comment]
@henit-chobisa UX changes [LGTM](https://en.wiktionary.org/wiki/LGTM)! I don't have strong feelings on the technical side, I'd suggest opening a PR with this and let someone from the `@gitpod-io/engineering-webapp` chime in with code suggestions or changes. 🏓

**Comment by henit-chobisa** [comment]
Sure @gtsiolis, I do the same, 
Also, the same `item component` that is used to represent `workspaces` here, is also used to represent `environment_variable list` and others, so should I change all of those or I make the PR with only the workspace change ?

**Comment by gtsiolis** [comment]
@henit-chobisa I'd suggest keeping the code changes to the minimum as a good [MVC](https://www.notion.so/gitpod/Core-Values-Differentiators-2ed4c2f93c84499b98e3b5389980992e#0c2e0d6c8a99404b97ccdc206b573d72), so we can review and merge faster. If I recall correctly the same component should be used in other pages.

**Comment by henit-chobisa** [comment]
Got it @gtsiolis, 
I will make the PR before EOD.Thank you so much  🙏

**Comment by henit-chobisa** [comment]
Hey @gtsiolis thank you so much for your support, 
I have made the PR, which fixes the Item List components for the workspace page, would make for other sections like `environment variables` and `Integrations` too, once the current PR method gets approved.
 
* Also, Can I tackle any issues from https://github.com/gitpod-io/gitpod/issues/12033 ?

**Comment by gtsiolis** [comment]
> I have made the PR

🙌 

> Can I tackle any issues from https://github.com/gitpod-io/gitpod/issues/12033 ?

@henit-chobisa The workspace list improvements are still in discussion. I'll post some updates later today or this week and ping you if you're up for making any changes. 🏓

**Comment by henit-chobisa** [comment]
Would be really really @gtsiolis , if I can have the opportunity to work on it. 🐳

**Comment by RayAsh37** [comment]
hey, @gtsiolis was going through the issue to understand #16420 when I came across this bug. The previous active workspace is never deactivated. I believe this isn't the intended behavior as confirmed here.

> to confirm we need to maintain the selection of the workspace item and the button corresponding to context menu till the time the drop down is active. I have already started working on it, will soon make a PR.

https://user-images.githubusercontent.com/70441807/234544070-585db26c-dc7e-486a-99e4-fe31d25b7813.mp4

**Comment by gtsiolis** [comment]
Nice catch, @RayAsh37! ⚾ 

Could you open a follow-up issue for this bug?
